### PR TITLE
Agent-196: Add asset for nmstateconfig

### PR DESCRIPTION
**Pull request by bfournie**
Add the asset for nmstateconfig and implement Load() to retrieve the YAML file (which may contain multiple nodes) and convert it the format suitable for NetworkManager.  The Generate() will be implemented later when the contents for the asset can be retrieved for another config file.

### CODE REVIEW

File: pkg/asset/agent/manifests/nmstateconfig_test.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+		fetchError         error
+		expectedFound      bool
+		expectedError      bool
+		requiresNmstatectl bool
+		expectedConfig     []*models.HostStaticNetworkConfig
+	}{
+		{
+			name: "valid-config-file",
+			data: `
+metadata:
```
Comment (andfasano) [inline pkg/asset/agent/manifests/nmstateconfig_test.go:37]: Not blocking: to improve the test readability, and to make it easier adding new cases, I'd recommend to follow the fluent interface approach for defining the actual scenario as described in [platform_test.go](https://github.com/openshift/installer/blob/3c4a5fb3d859cc150a2dd936517501ba07d5e825/pkg/types/baremetal/validation/platform_test.go#L20)

File: pkg/asset/agent/manifests/nmstateconfig.go [hunk 2]
```diff
@@ -1 +1,152 @@
...
+// Read a YAML file containing multiple YAML definitions of the same format
+// Each specific format must be of type DecodeFormat
+func getMultipleYamls(contents []byte, decoder decodeFormat) ([]interface{}, error) {
+
+	r := bytes.NewReader(contents)
+	dec := yaml.NewYAMLToJSONDecoder(r)
+
+	var outputList []interface{}
+	for {
+		decodedData, err := decoder.NewDecodedYaml(dec)
```
[the remark this answers is not in the document]
Comment (bfournie) [reply pkg/asset/agent/manifests/nmstateconfig.go:126]: This yaml file has multiple NMStateConfig yamls in it so we can't use yaml.Unmarshal() directly, which is why the Decode is done, as it has to take the type e.g. https://pkg.go.dev/gopkg.in/yaml.v2#Decoder.Decode.

File: pkg/asset/agent/manifests/nmstateconfig.go [hunk 3]
```diff
@@ -1 +1,152 @@
...
+	}
+
+	// Split up the file into multiple YAMLs if it contains NMStateConfig for more than one node
+	var decoder nmStateConfigYamlDecoder
+	yamlList, err := getMultipleYamls(file.Data, &decoder)
+	if err != nil {
+		return false, errors.Wrapf(err, "could not decode YAML for %s", nmStateConfigFilename)
+	}
+
+	var staticNetworkConfig []*models.HostStaticNetworkConfig
```
[the remark this answers is not in the document]
Comment (bfournie) [reply pkg/asset/agent/manifests/nmstateconfig.go:79]: Yes, good point. Will be good for a follow-up, I won't make any changes to this PR for it.

File: pkg/asset/agent/manifests/nmstateconfig_test.go (hunk 1 above)
Comment (bfournie) [reply pkg/asset/agent/manifests/nmstateconfig_test.go:37]: Per our discussion, yes that this is a good idea for readability. As the tests are consistent with the other agent manifests (infraenv_test.go, pullsecret_test.go) I'd like to keep it as is for this PR and do the refratoring later

**Pull request merged**